### ISSUE TITLE
[UX] Additional info on search engines in preferences/engines

### ISSUE THREAD

**Issue by ghost** [searx/searx#2442]
**Is your feature request related to a problem?**
I discovered a new engine on the search engine settings page (for example: naver) ­– so I had to do an extra search on it to determine _what it was and if it might be relevant to my use case_.

**Describe the solution you'd like**
A short descripton (introtext of its Wikipedia page or just the descripton from Wikidata) of the search engine.
Maybe even a specific template for all engines on additional info, like: open|closed source, own index|meta search of Bing+Yandex, locality-specific|worldwide, server country, etc.
…when the engine list grows it might also be helpful to categorize (open|closed source or so) and rank (on popularity or so) on the settings page for easier overlook.

**Describe alternatives you've considered**
I've had to do an extra search on the search engines name.

I'd love to read **other opinions** on the issue and **possible implementations** (especially pertaining to user experience)!

**Comment by unixfox** [comment]
Good feature request but may I ask you how would you implement that in the Searx interface?

Where should the description appear? When moving the mouse cursor over the engine name in the preferences?

Ideally we would like to find a way so that it wouldn't require adding JavaScript code in order to make it compatible with and without JavaScript enabled.

**Comment by dalf** [comment]
I suggest to modify the searx source of the engines. For example with `searx/engines/github.py` :
https://github.com/searx/searx/blob/14a395a6718949015e53a0b593eaa2f28e43fd27/searx/engines/github.py#L1-L17

Becomes : 
```python
"""
Github (It)

"""

from json import loads
from urllib.parse import urlencode

# about
about = {
    "website": 'https://github.com/',
    "wikidata_id": 'Q364',
    "official_api_documentation": 'https://developer.github.com/v3/',
    "results": 'JSON',
    "use_official_api": True,
    "require_api_key": False,
}
...
```

From [Q364](https://www.wikidata.org/wiki/Q364), a [script](https://gist.github.com/dalf/db85f103de9c28b5fcb7896525a6d41b#file-fetch_engine_descriptions-py) can fetch the engine descriptions in different languages.

See the [result here](https://gist.github.com/dalf/db85f103de9c28b5fcb7896525a6d41b#file-output-json) (173KB gzip compressed).

All these information can be displayed on a tooltip in the preferences.

But, some engine don't have a wikidata page.

The straightforward solution doesn't match the `wikidata_id` idea: 
```
about = {
    "website": 'https://github.com/',
    "description": gettext('GitHub, Inc. is a subsidiary of Microsoft which provides hosting for software development and version control using Git. It offers the distributed version control and source code management (SCM) functionality of Git, plus its own features. It provides access control and several collaboration features such as bug tracking, feature requests, task management, continuous integration and wikis for every project. Headquartered in California, it has been a subsidiary of Microsoft since 2018.'),
    "official_api_documentation": 'https://developer.github.com/v3/',
    "results": 'JSON',
    "use_official_api": True,
    "require_api_key": False,
}
...
```

* `gettext` means the translation is managed by transifex
* `wikidata_id` means the translation come from wikidata / wikipedia without edition possibility.


<details><summary>Python to modify the searx code automatically</summary>
<p>

A gist that may help : https://gist.github.com/dalf/db85f103de9c28b5fcb7896525a6d41b 
* copy the two files at the root directory, where `Makefile` is
* `make pyenvinstall`
* `. ./local/py3/bin/activate` (requires to use the `searx/engines/wikidata.py`, see below )
* `pip install redbaron`
* `python rewrite.py`  modify the searx code

The wikidata_id is found automatically using the engine URL using a request like this : https://w.wiki/tRu

If https://www.wikidata.org/wiki/Q364 is shown to the user, then the user: 
* can choose a wikipedia page
* check the various wikidata properties

Searx could also fetch automatically (either at runtime, either with an external script like `searx/languages/py`):
* the wikipedia sumaries from the `Q364`.
* the logo
* (all other wikdata properties, but summary and logo should be enough)

So with this change, searx can display in the preferences the different information (and helth information with the checker).

Important: the gist change automatically the searx source code for all the `searx/engines/*.py` file, but not for the engines declare only in `searx/settings.yml`.

---

For reference: https://github.com/searx/searx/issues/302#issuecomment-325743059

</p>
</details>

**Comment by ghost** [comment]
Display engine descripton: While my initial thought also was to show a quickinfo on hover over the engine name, that would be very unintuitive for the **mobile** experience.
So maybe right next to the engine name, or a toggable "i" icon that shows/hides the descripton.

For the _overview/sorting experience_ I had imagined something like searx.space or a Wikipedia chart (sortable when clicked on a column header) – I understand though that this is impossible without JavaScript…
So the **default** sorting (preferably adjustable by the instance maintainer) would be important for UX too - I'd suggest sorting it by _popularity_ (Google,Wikipedia,Bing,DuckDuckGo,etc.)
I don't see a reason why the engine quantity won't increase - so, for example the engines/general chart could be subdivided for right now into 'general', 'meta search', 'language' (and those could be toggable to show/hide).

Also consider adding more info to the chart **to help decide what to use** like I mentioned in the first post (like: open|closed source, own index|meta search (of Bing+Yandex), locality-specific|worldwide, server country, etc.).

(Unfortunately I can't code, so somebody else motivated would have to realise any implementation… already thank you @dalf )

**Comment by return42** [comment]
@gnu1dea I guess this issue can be closed since #2661 is merged / right?
